Src/DependencyCollector/Shared.Tests/Implementation/FrameworkHttpProcessingTest.cs

```diff
@@ -51,6 +51,12 @@ public void Cleanup()
 
         #region BeginEndCallBacks
 
+        [TestMethod]
+        public void OnBeginDoesNotThrowForIncorrectUrl()
+        {
+            this.httpProcessingFramework.OnBeginHttpCallback(100, "BadUrl"); // Should not throw
```

Comment:
Yes, no it has try/catch. In general we should never throw and crash customer app.